### PR TITLE
Only keep the mutex lock for the duration of setting the variable

### PR DESCRIPTION
**Pull request by dmathieu**
Acquiring the connection can take some time, which causes timeouts.

### CODE REVIEW

**Comment by tt** [comment]
How about using [`atomic.AddInt64`](https://golang.org/pkg/sync/atomic/#AddInt64) instead?

File: broker/redis.go [hunk 1]
```diff
@@ -28,12 +28,23 @@ type pool struct {
 }
 
 func (p *pool) Get() Conn {
+	n := p.incCounter()
```
Comment (tt) [inline broker/redis.go]: I think this would be `n := atomic.AddInt64(&p.c, 1)` using that package.

**Comment by tt** [comment]
This obviously makes sense to do but why would we wait a long time to get a connection? Is the pool not configured appropriately for busl?

File: broker/redis.go (hunk 1 above)
Comment (dmathieu) [inline broker/redis.go]: Nice, I didn't know about that. Changed.

**Comment by dmathieu** [comment]
The pool opens connections as needed, apparently starting with a single one.
So it would make sense that it starts by opening a lot of them, which definitely causes issues if it can't happen concurrently.

**Pull request merged**